lib/ClangImporter/ImportType.cpp

```diff
@@ -2203,6 +2203,47 @@ static Type decomposeCompletionHandlerType(
   }
 }
 
+ImportedType ClangImporter::Implementation::importPropertyType(
+                                        const clang::ObjCMethodDecl *decl,
+                                        DeclContext *dc,
+                                        importer::ImportedName name,
+                                        bool isFromSystemModule) {
+  // here we expect a method that is being imported as an effectful property.
+  // thus, we currently require async info.
+  if (!name.getAsyncInfo()) {
+    return ImportedType();
+  }
+
+  if (decl->isVariadic()) {
+    return ImportedType();
+  }
+
+  // Our strategy here is to determine what the return type of the method would
+  // be, had we imported it as a method.
+
+  // these are ignored, because we only care about the return type here.
+  Optional<ForeignAsyncConvention> asyncConvention;
+  Optional<ForeignErrorConvention> errorConvention;
+
+  const auto kind = SpecialMethodKind::Regular;
+
+  // Import the parameter list and result type.
+  ParameterList *bodyParams = nullptr;
+  ImportedType importedType;
+
+  auto methodReturnType = importMethodParamsAndReturnType(
+      dc, decl, decl->parameters(), false,
+      isFromSystemModule, &bodyParams, name,
+      asyncConvention, errorConvention, kind);
+
+  // getter mustn't have any parameters!
+  if (bodyParams->size() != 0) {
+    return ImportedType();
+  }
+
+  return methodReturnType;
```

Comment:
I think you should check to ensure that we have an async convention but no error convention, and bail out if either of those doesn't hold:

```
if (!asyncConvention || errorConvention)
  return ImportedType();
```